content/docs/atom-guide/dev-guide/shaders/azsl/_index.md

```diff
@@ -1,20 +1,468 @@
+
+* [Shader Resource Groups](#ShaderResourceGroup).
+* [Root Constants](#rootconstant).
+* `option` variables, but these variables can also be used equivalently as Vulkan's Specialization Constants, in other words, these constants can be statically defined at compilation time to avoid branching at runtime.  
+  
+**AZSLc** will trigger a compilation error if an arbitrary `cbuffer` or any other type of resources are declared outside the scopes mentioned above. The exception would be `static` global variables, which won't be exposed to the application but will be usable internally by the shaders.
+
+## <a name="ShaderResourceGroup"></a>Shader Resource Groups
+
+A *Shader Resource Group (SRG)* is a logical namespace to declare shader constants and resources. In AZSL the keyword to declare an SRG is `ShaderResourceGroup`. Shader resources are constants of the type Shader Resource Views (SRV), Unordered Access Views (UAV), Constant Buffer Views (CBV), and Sampler States.
+
+When compiling shaders with the "--use-spaces" argument, each SRG will take over a whole register space (or descriptor set, as known in Vulkan). AZSL abstracts the register and register space assignment of resource descriptors with the `ShaderResourceGroupSemantic` keyword, which must be specified when declaring SRGs.
+
+A Shader Resource Group is declared with the following syntax:
+  
+    ShaderResourceGroup <Name> : <Semantic>
+    {
+        <Data>
+    };
+     
+    //Alternatively
+    partial ShaderResourceGroup <Name> : <Semantic>
+    {
+        <Data>
+    };
+  
+* **&lt;Name&gt;** is the user defined name for the SRG.
+* **&lt;Semantic&gt;** is the name of the `ShaderResourceGroupSemantic` for the SRG. More on this later.
+* **&lt;Data&gt;** is one or more declaration of functions, constants and resources for the binding data contained in the SRG.  
+  
+In the code block shown above, a [partial](#partial) qualifier can be used when declaring SRGs.
+### &lt;Data&gt;
+SRGs can contain fundamental type variables like float, float3, int2, bool, matrices, etc. For each SRG, a Constant Buffer View (CBV) will be automatically created and it will pack all the fundamental type variables declared inside the SRG.
+
+SRGs can contain resource declarations like Texture2D, Texture2DMS, ByteAddressBuffer, etc. In general any HLSL shader resource of the categories:  Shader Resource Views (SRV), Unordered Access Views (UAV), Samplers and Constant Buffer Views (CBV).
+
+SRGs can contain unbounded arrays for resource declarations: SRVs, UAVs, CBVs and Samplers. There are some rules that need to be considered, depending on command line arguments when compiling with **AZSLc**. More on unbounded arrays [here](unbounded-arrays).
+
+SRGs can contain functions.
+
+SRGs can contain structs (declaration & instances). The Shader Build Pipeline and the runtime handle the platform-specific packing rules and alignment.
```

Comment:
```suggestion
- Structs (declaration and instances). The Shader Build Pipeline and runtime handle the struct's platform-specific packing rules and alignment.
```

- Is Shader Buid Pipeline a proper noun? If so, remove "The" before it and capitalize "Shader Build Pipeline" in other places in your docs.
- How does the second sentence relate to structs? Is it the struct's platform-specific packing rules and alignment? I'm not sure what this means and why it's relevant for the user to know this.

Reply:
Fixed.